ui/src/app/machines/components/HeaderStrip/HeaderStrip.js

```diff
@@ -1,107 +1,29 @@
+import { machine as machineActions } from "app/base/actions";
+import { machine as machineSelectors } from "app/base/selectors";
+import ActionFormWrapper from "./ActionFormWrapper";
+import AddHardwareMenu from "./AddHardwareMenu";
+import HeaderStripTabs from "./HeaderStripTabs";
+import TakeActionMenu from "./TakeActionMenu";
 
-const getAddHardwareLinks = navigationOptions => {
-  const links = [
-    {
-      children: "Machine",
-      element: Link,
-      to: "/machines/add"
-    },
-    {
-      children: "Chassis",
-      element: Link,
-      to: "/machines/chassis/add"
-    }
-  ];
-
-  return navigationOptions.rsd
-    ? links.concat({
-        children: "RSD",
-        element: "a",
-        href: `${process.env.REACT_APP_ANGULAR_BASENAME}/rsd`
-      })
-    : links;
-};
-
-const getTakeActionLinks = (actionOptions, machines) => {
-  const initGroups = [
-    { type: "lifecycle", items: [] },
-    { type: "power", items: [] },
-    { type: "testing", items: [] },
-    { type: "lock", items: [] },
-    { type: "misc", items: [] }
-  ];
-
-  const groupedLinks = actionOptions.reduce((groups, option) => {
-    const count = machines.reduce((sum, machine) => {
-      if (machine.actions.includes(option.name)) {
-        sum += 1;
-      }
-      return sum;
-    }, 0);
-
-    if (count > 0 || option.type === "lifecycle") {
-      const group = groups.find(group => group.type === option.type);
-      group.items.push({
-        children: (
-          <div className="u-flex-between">
-            <span data-test={`action-title-${option.name}`}>
-              {option.title}
-            </span>
-            {machines.length > 1 && (
-              <span
-                data-test={`action-count-${option.name}`}
-                style={{ marginLeft: ".5rem" }}
-              >
-                {count || ""}
-              </span>
-            )}
-          </div>
-        ),
-        disabled: count === 0,
-        onClick: () => null
-      });
-    }
-    return groups;
-  }, initGroups);
-
-  return groupedLinks.map(group => group.items);
-};
-
-export const HeaderStrip = ({ selectedMachines }) => {
+export const HeaderStrip = ({ selectedMachines, setSelectedMachines }) => {
   const dispatch = useDispatch();
   const machines = useSelector(machineSelectors.all);
   const machinesLoaded = useSelector(machineSelectors.loaded);
-  const resourcePools = useSelector(resourcePoolSelectors.all);
-  const resourcePoolsLoaded = useSelector(resourcePoolSelectors.loaded);
-  const actionOptions = useSelector(generalSelectors.machineActions.get);
-  const navigationOptions = useSelector(generalSelectors.navigationOptions.get);
-  const { location } = useRouter();
+
+  const [selectedAction, setSelectedAction] = useState();
 
   useEffect(() => {
-    dispatch(generalActions.fetchMachineActions());
-    dispatch(machineActions.fetch());
-    dispatch(poolActions.fetch());
-  }, [dispatch]);
+    if (!machinesLoaded) {
```

Comment:
Again, this should be handled by the saga, if not we should fix that.

Reply:
removed